src/node_jsvmapi.cc

```diff
@@ -23,6 +23,24 @@
 #include <vector>
 #include <string.h>
 
+static v8::Persistent<v8::Value> *lastException = 0;
+#define NAPI_PROLOGUE \
+  v8::TryCatch napi_internal_tryCatch(v8::Isolate::GetCurrent())
+
+#define NAPI_RETURN(returnExpression) \
```

Comment:
>NAPI_RETURN [](start = 8, length = 11)

Could this be implemented as a class with a destructor instead of a macro? Then instead of changing all the `return` statements, the destructor would do this work when the instance goes out of scope.